src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -116,7 +116,7 @@ void QgsLayoutItemLegend::paint( QPainter *painter, const QStyleOptionGraphicsIt
 
   if ( mLayout )
   {
-    if ( !mLayout->renderContext().isPreviewRender() && mLegendModel->hitTestInProgress() )
+    if ( mLegendModel->hitTestInProgress() )
     {
       mLegendModel->waitForHitTestBlocking();
     }
```

Comment:
Shall we not remove the whole block.

My understanding is that we want to:
- in preview, use thread and don't wait at all to not block UI (the legend/paint would be refreshed when the hit test task is completed)
- on any other "mode", we don't use thread, and so there is no need to wait!

Prior to this modification, we were using thread for non-preview mode, so we needed to wait, but that's not the case anymore. Am I right?

Reply:
Hm good catch -- I accidentally inverted the logic here!

Reply:
Fixed